v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -287,12 +289,23 @@ private static TableConfig getTableConfig(
      */
     if (indexInfo.containsKey(tableName)) {
       ImmutableList<SourceColumnIndexInfo> tableIndexInfo = indexInfo.get(tableName);
-
       // TODO(vardhanvthigle): support for non-primary indexes.
-      tableIndexInfo.stream()
-          .filter(info -> info.isPrimary() && info.ordinalPosition() == 1)
-          .map(SourceColumnIndexInfo::cardinality)
-          .forEach(tableConfigBuilder::setApproxRowCount);
+      Optional<Long> tableCardinality =
+          tableIndexInfo.stream()
+              .filter(info -> info.isPrimary() && info.ordinalPosition() == 1)
+              .map(SourceColumnIndexInfo::cardinality)
+              .findFirst();
+
+      tableCardinality.ifPresentOrElse(
+          tableConfigBuilder::setApproxRowCount,
+          () -> {
+            throw new PrimaryKeyNotFoundException(
+                new Throwable(
+                    String.format(
+                        "Migration of tables with no primary keys is not supported!"
```

Comment:
This might come up even if there are primary keys which are not supported. We should tweak the error message here.

Reply:
From the code it seems like it the indexes list is being reduced to find only the column which is in the first ordinal position of the primary key. 

A false on that check should mean that there is no column in the list retrieved that has `isPrimary()` AND `ordinalPosition = 1`. If there is in fact a PK, this condition will always be true.

Hence it should mean it will never come up if there is a primary key. What am I missing?

Please also suggest the error message to tweak nonetheless.